src/flint/flint_base/flint_base.pyx

```diff
@@ -265,22 +264,63 @@ cdef class flint_poly(flint_elem):
         raise NotImplementedError("Complex roots are not supported for this polynomial")
 
 
+class Ordering(enum.Enum):
```

Comment:
Not an urgent issue but defining this here and not in the .pxd means that the type cannot be imported with `cimport` and used as the type declaration on `cdef` functions.

Reply:
I'm not sure how that would be achieved, though I might just be missing something. 
A definition like 
```python
class Ordering(enum.Enum):
    lex = "lex"
    deglex = "deglex"
    degrevlex = "degrevlex"
```
cannot go in the pxd file. Similarly neither can an empty definition like 
```python
class Ordering(enum.Enum):
    pass
```
The class can't be declared as a `cdef class` because `enum.Enum` is not an extension type, which is also why the `ordering_py_to_c` and `ordering_c_to_py` functions are not members. This was why it was a cython enum previously.

Cython enums don't support string values, and it complains that a byte literal with more than one char isn't coercible to an integer, fair enough, C enums are only ints anyway. With a single byte the Cython compiler crashes.

We could go back to using a Cython cpdef enum with int values, but then using it within a type hint raises a very unhelpful error message when passed something that's not an int, "TypeError: an integer is required". The range of the enum isn't even enforced.
```python
# in a .pyx
cpdef enum CheeseState:
    hard = 1
    soft = 2
    runny = 3

def cheese(arg: CheeseState):
    print(arg)
```
```python
>>> cheese(1)
1
>>> cheese(100000)
100000
>>> cheese("test")
...
TypeError: an integer is required
```

Reply:
We could ofc just do it ourselves with string values and functions to go between

Reply:
Oh, I see it is a subclass of a Python class.

The whole enum business seems unnecessarily confusing in Python and Cython.